src/HordeString.php

```
@@ -953,4 +953,46 @@ public static function trimUtf8Bom($str)
             ? substr($str, 3)
             : $str;
     }
+
+    /**
+     * Try to convert a string to UTF-8.
+     * Returns null if source charset could not be detected or conversion failed
+     *
+     * @param string $string The string to convert
+     * @return null|string The string in UTF-8 or null if conversion failed
+     */
+    public static function convertToUtf8(string $string): ?string
+    {
+        $encodings = [
+            'UTF-8',
+            'ISO-8859-1',
+            'UTF-16',
+            'ISO-8859-2',
+            'ISO-8859-3',
+            'ISO-8859-4',
+            'ISO-8859-5',
+            'ISO-8859-6',
+            'ISO-8859-7',
+            'ISO-8859-8',
+            'ISO-8859-9',
+            'ISO-8859-10',
+            'ISO-8859-13',
+            'ISO-8859-14',
+            'ISO-8859-15',
+            'ISO-8859-16',
+        ];
+        $enc = mb_detect_encoding($string, $encodings);
+        if (!$enc) {
+            return null;
+        } elseif ($enc === 'UTF-8') {
+            return $string;
+        }
+        $converted = iconv($enc, "UTF-8", $string);
+        if ($converted === false) {
+            return null;
```

Instead of returning null, I think it would be better to throw exceptions with a descriptive error message. This would make it easier to spot cases, where a) the encoding could not be detected and b) the conversion failed.

---

Ok I changed that. I also added a check to see if iconv is installed

---

Thank Good catch with the iconv-check!